pkg/framework/webhooks.go

```diff
@@ -0,0 +1,154 @@
+			return true, nil
+		} else if err != nil {
+			klog.Errorf("error querying api for ValidatingWebhookConfiguration object %q: %v, retrying...", webhookConfiguraiton.Name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+// DeleteMutatingWebhookConfiguration deletes the specified MutatingWebhookConfiguration object
+func DeleteMutatingWebhookConfiguration(c client.Client, webhookConfiguraiton *admissionregistrationv1.MutatingWebhookConfiguration) error {
+	return wait.PollImmediate(RetryShort, WaitShort, func() (bool, error) {
+		if err := c.Delete(context.TODO(), webhookConfiguraiton); apierrors.IsNotFound(err) {
+			return true, nil
+		} else if err != nil {
+			klog.Errorf("error querying api for MutatingWebhookConfiguration object %q: %v, retrying...", webhookConfiguraiton.Name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+// UpdateMutatingWebhookConfiguration updates the specified mutating webhook configuration
+func UpdateMutatingWebhookConfiguration(c client.Client, updated *admissionregistrationv1.MutatingWebhookConfiguration) error {
+	return wait.PollImmediate(RetryShort, WaitMedium, func() (bool, error) {
+		existing, err := GetMutatingWebhookConfiguration(c, updated.Name)
+		if err != nil {
+			klog.Errorf("Error getting MutatingWebhookConfiguration: %v", err)
+			return false, nil
+		}
+		if err := c.Patch(context.TODO(), existing, client.MergeFrom(updated)); err != nil {
+			klog.Errorf("error patching MutatingWebhookConfiguration object %q: %v, retrying...", updated.Name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+// UpdateValidatingWebhookConfiguration updates the specified mutating webhook configuration
+func UpdateValidatingWebhookConfiguration(c client.Client, updated *admissionregistrationv1.ValidatingWebhookConfiguration) error {
+	return wait.PollImmediate(RetryShort, WaitMedium, func() (bool, error) {
+		existing, err := GetValidatingWebhookConfiguration(c, updated.Name)
+		if err != nil {
+			klog.Errorf("Error getting ValidatingWebhookConfiguration: %v", err)
+			return false, nil
+		}
+		if err := c.Patch(context.TODO(), existing, client.MergeFrom(updated)); err != nil {
+			klog.Errorf("error patching ValidatingWebhookConfiguration object %q: %v, retrying...", updated.Name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+// IsMutatingWebhookConfigurationSynced expects a matching MutatingWebhookConfiguration to be present in the cluster
+func IsMutatingWebhookConfigurationSynced(c client.Client) bool {
+	if err := wait.PollImmediate(RetryShort, WaitLong, func() (bool, error) {
+		existing, err := GetMutatingWebhookConfiguration(c, DefaultMutatingWebhookConfiguration.Name)
+		if err != nil {
+			klog.Errorf("Error getting MutatingWebhookConfiguration: %v", err)
+			return false, nil
+		}
+
+		equal := equality.Semantic.DeepDerivative(DefaultMutatingWebhookConfiguration.Webhooks, existing.Webhooks)
```

Comment:
Might be worth a note that this will should the CABundle field that has been set by the injector and that's why we are using `DeepDerivative`